docs/coding-guidelines.md

```diff
@@ -370,6 +372,14 @@ FormatString
 GetData
 ```
 
+The following test names are acceptable, but not preferred:
+
+```cs
+PublicApiArgumentsShouldNotBeNull
+```
```

Comment:
What's the point in having divergent guidance? I say remove this

Reply:
In the interest of making this less intrusive. I'd be happy to remove it. Can anyone else chime in @NuGet/nuget-client

Reply:
I'm with Donnie on this. If we want all our tests to eventually use the proposed format (with the `_` separators), then don't mention that there's an allowed exception to the "rule". It will just be used as an excuse to never change.

Reply:
My goal was to clarify that we want a 3 part name.
The `_` in my opinion just adds to the readability. 

I'll remove this statement and move the example to the not acceptable list.